airflow/providers/cloudera/hooks/cde_hook.py

```diff
@@ -0,0 +1,375 @@
+
+        except SecurityError as err:
+            self.log.error(
+                "Failed to get the cde auth token for the connection %s, error: %s",
+                self.cde_conn_id,
+                err,
+            )
+            raise CdeHookException(err) from err
+
+        return cde_token
+
+    def submit_job(
+        self,
+        job_name: str,
+        variables: dict[str, Any] | None = None,
+        overrides: dict[str, Any] | None = None,
+        proxy_user: str | None = None,
+    ) -> int:
+        """
+        Submit a job run request
+
+        :param job_name: The name of the job definition to run (should already be
+            defined in the virtual cluster).
+        :param variables: Runtime variables to pass to job run
+        :param overrides: Overrides of job parameters for this run
+        :return: the job run ID for a successful submission or an AirflowException
+        :rtype: int
+        """
+        if proxy_user:
+            # Not tested because it should have never been introduced in the first place
+            self.log.warning("Proxy user is not yet supported. Setting it to None.")  # pragma: no cover
+
+        body = dict(
+            variables=variables,
+            overrides=overrides,
+            # Shall be updated to proxy_user when we support this feature
+            user=None,
+        )
+        response = self._do_api_call("POST", f"/jobs/{job_name}/run", body)
+        if response is None:
+            msg = f"Unexpected 'None' response for '{job_name}' job."
+            self.log.error(msg)
+            raise CdeHookException(msg=msg)
+        try:
+            job_run_id: int = response["id"]
+        except KeyError as err:
+            msg = f"Response for '{job_name}' job do not contain 'id'."
+            raise CdeHookException(err, msg) from err
+        return job_run_id
+
+    def kill_job_run(self, run_id: int) -> None:
+        """
+        Kill a running job
+
+        :param run_id: the run ID of the job run
+        """
+        self._do_api_call("POST", f"/job-runs/{run_id}/kill")
+
+    def check_job_run_status(self, run_id: int) -> str:
+        """
+        Check and return the status of a job run
+
+        :param run_id: the run ID of the job run
+        :return: the job run status
+        :rtype: str
+        """
+        response = self._do_api_call("GET", f"/job-runs/{run_id}")
+        if response is None:
+            msg = f"Unexpected 'None' response for '{run_id}' job run."
+            self.log.error(msg)
+            raise CdeHookException(msg=msg)
+
+        try:
+            response_status: str = response["status"]
+        except KeyError as err:
+            msg = f"Response for '{run_id}' job run do not contain 'status'."
+            raise CdeHookException(err, msg) from err
+        return response_status
+
+    def get_conn(self):  # pylint: disable=missing-function-docstring; not required for CdeHook
+        raise NotImplementedError
```

Comment:
Seems like it is a good idea to move obtain connection to this method instead of get it in class constructor

```python
        airflow_connection = self.get_connection(self.cde_conn_id)
        self.connection = CdeConnection.from_airflow_connection(airflow_connection)
```